PocketKit/Sources/SaveToPocketKit/SaveToAddTagsViewModel.swift

```diff
@@ -0,0 +1,29 @@
+import Combine
+import Sync
+import Textile
+
+class SaveToAddTagsViewModel: AddTagsViewModel {
+    private let item: SavedItem?
+    private let retrieveAction: ([String]) -> [Tag]?
+    private let saveAction: ([String]) -> Void
```

Comment:
What's the thinking behind having these as functions that are injected? An immediate thought is so that we don't have to further pass down dependencies. Perhaps this makes testing easier, too? Just curious.

Reply:
These functions ultimately calls methods in the `PocketSaveService` which the `SavedItemViewModel` already depends on and so I felt that it was simpler to utilize the save service in that view model instead of injecting it in the `SaveToAddTagsViewModel`. I personally think it makes testing easier too for `SaveToAddTagsViewModel` since we don't need to consider the save service + stub the methods, though we do have a mock for it.

I didn't do this same type of implementation with `source` for PocketAddTagsViewModel since I didn't think of this way until this PR but I can if you recommend to be consistent. Is there a way / implementation you prefer? Thanks for asking this - love the curious questions 😄

Reply:
It seems akin to the factory pattern (though we're not injecting a dependency), which is something I don't have a lot of "experience" with. @jacobsimeon is more well-versed in it than me, so I'd be interested what his $0.02 here are. But, nothing blocking, just a curiosity.

Reply:
okay! I do have a tech debt ticket to refactor parts of the tag work, so we can revisit the discussion then! thanks @dskuza

Reply:
The pattern @cyndichin is using seems like an alternative to a) having a delegate that can respond to certain actions or b) injecting in another dependency like `source`. I see Cyndi's point that it makes testing easier since you don't have to inject a mock source. That said, the actual method being called on source is a critical piece of behavior in the case of saving tags.
So, I'd actually prefer to see `source` getting injected here and then have a test on `SaveToAddTagsViewModel` that asserts we're calling the correct methods. Though, it does occur to me that it would get complicated since we're in the context of the save extension.

Not saying anything needs to change here, but I do prefer passing in the object(s) you need (`source` or `saveService` in this case) to do work instead of passing in some closures that happen to be easier to construct when initializing the view model.

That said, it would be tricky to have this view model take in a single dependency since the main app wants to use `Source` and the extension wants to use `SaveService`. So you'd end up having to create another protocol that can be conformed to by both `Source` and `SaveService` and injecting that.

Again, not saying anything needs to change (this is a perfectly fine solution), but you asked for my $0.02. I guess I gave $200 though 😆